arangod/Agency/RestAgencyHandler.cpp

```diff
@@ -181,13 +181,22 @@ RestStatus RestAgencyHandler::pollIndex(
             builder.add("code", VPackValue(int(ResponseCode::OK)));
             builder.add(VPackValue("result"));
             VPackObjectBuilder r(&builder);
-            auto const firstIndex = slice.get("firstIndex");
+            if (!slice.get("firstIndex").isNumber()) {
+              generateError(
+                rest::ResponseCode::SERVER_ERROR,
+                TRI_ERROR_HTTP_SERVER_ERROR, "invalid first log index.");
+            } else if (slice.get("firstIndex").getNumber<uint64_t>() > start) {
+              generateError(
+                rest::ResponseCode::SERVER_ERROR,
+                TRI_ERROR_HTTP_SERVER_ERROR, "first log index is greater than requested.");
+            }
+            uint64_t i = slice.get("firstIndex").getNumber<uint64_t>() - start;
```

Comment:
We have just made sure that start >= firstIndex, now we should better do the difference the other way round.

Reply:
done